Failing require() in configuration file causes undescriptive error

Bug report
---

- Node Version: `4.4.5`
- Protractor Version: `3.3.0`
- Operating System and Version `Windows 10`

In my configuration file I had a dependency, which I imported with `require('modulename')`, but after updating packages.json this broke the call. However, Protractor responds with the following output:

```
[10:33:08] E/configParser - error code: 105
[10:33:08] E/configParser - description: failed loading configuration file C:\path\protractor.conf.js
```

As hinted per [this StackOverflow remark](http://stackoverflow.com/questions/37000131/protractor-error-105-when-trying-to-run-spec-js#comment63229521_37000131), adding `logger.error(e);` to https://github.com/angular/protractor/blob/master/lib/configParser.ts#L200, revealed the actual error:

```
[10:36:29] E/configParser - { [Error: Cannot find module 'modulename'] code: 'MODULE_NOT_FOUND' }
```

These parsing errors should not be masked by the current error message and should also be reported in the output.

This will be for milestone 4.0.0. So it'll be released soon.

@bgever how's this for a stack trace?

```
[10:42:33] E/configParser - Error code: 105
[10:42:33] E/configParser - Error message: failed loading configuration file foobar.js
[10:42:33] E/configParser - Error: Cannot find module '/Users/craignishina/src/protractor/foobar.js'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at ConfigParser.addFileConfig (/Users/craignishina/src/protractor/built/configParser.js:127:26)
    at Object.<anonymous> (/Users/craignishina/src/protractor/spec/unit/configParser_test.js:25:16)
    at attemptSync (/Users/craignishina/src/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1886:24)
    at QueueRunner.run (/Users/craignishina/src/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1874:9)
    at QueueRunner.execute (/Users/craignishina/src/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1859:10)
    at Spec.Env.queueRunnerFactory (/Users/craignishina/src/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:697:35)
```

--- a/lib/configParser.ts
+++ b/lib/configParser.ts
@@ -198,7 +198,7 @@ export class ConfigParser {
       fileConfig = require(filePath).config;
     } catch (e) {
       throw new ConfigError(
-          logger, 'failed loading configuration file ' + filename);
+          logger, 'failed loading configuration file ' + filename, e);
     }
     if (!fileConfig) {
       throw new ConfigError(

--- a/lib/exitCodes.ts
+++ b/lib/exitCodes.ts
@@ -22,11 +22,17 @@ export class ProtractorError extends Error {
     (Error as any).captureStackTrace(this, this.constructor);
   }
 
-  constructor(logger: Logger, message: string, code: number) {
+  constructor(logger: Logger, message: string, code: number, error?: Error) {
     super(message);
     this.message = message;
     this.code = code;
     this.captureStackTrace();
+
+    // replacing the stack trace with the thrown error stack trace.
+    if (error) {
+      let protractorError = error as ProtractorError;
+      this.stack = protractorError.stack;
+    }
     this.logError(logger);
 
     if (!ProtractorError.SUPRESS_EXIT_CODE) {
@@ -54,8 +60,8 @@ export class ProtractorError extends Error {
  */
 export class ConfigError extends ProtractorError {
   static CODE = CONFIG_ERROR_CODE;
-  constructor(logger: Logger, message: string) {
-    super(logger, message, ConfigError.CODE);
+  constructor(logger: Logger, message: string, error?: Error) {
+    super(logger, message, ConfigError.CODE, error);
   }
 }
 

--- a/spec/unit/configParser_test.js
+++ b/spec/unit/configParser_test.js
@@ -26,6 +26,7 @@ describe('the config parser', function() {
       } catch (err) {
         errorFound = true;
         expect(err.code).toEqual(ConfigError.CODE);
+        expect(err.stack).toMatch('Cannot find module');
       }
       expect(errorFound).toBe(true);
     });
@@ -38,6 +39,7 @@ describe('the config parser', function() {
       } catch (err) {
         errorFound = true;
         expect(err.code).toEqual(ConfigError.CODE);
+        expect(err.stack).toMatch('did not export a config object');
       }
       expect(errorFound).toBe(true);
     });
@@ -52,6 +54,7 @@ describe('the config parser', function() {
       } catch (err) {
         errorFound = true;
         expect(err.code).toEqual(ConfigError.CODE);
+        expect(err.stack).toMatch('Unknown test suite: foo.js');
       }
       expect(errorFound).toBe(true);
     });

